service/src/java/org/apache/hive/service/cli/CLIService.java

```diff
@@ -457,9 +469,34 @@ public OperationStatus getOperationStatus(OperationHandle opHandle)
     }
     OperationStatus opStatus = operation.getStatus();
     LOG.debug(opHandle + ": getOperationStatus()");
+    opStatus.setJobProgressUpdate(progressUpdateLog(getProgressUpdate, operation));
     return opStatus;
   }
 
+  private JobProgressUpdate progressUpdateLog(boolean isProgressLogRequested, Operation operation) {
+    if (isProgressLogRequested && canProvideProgressLog()) {
+        if (OperationType.EXECUTE_STATEMENT.equals(operation.getType())) {
+          SessionState sessionState = operation.getParentSession().getSessionState();
+          try {
+            while (sessionState.progressMonitor() == null && !operation.isFinished()) {
+              Thread.sleep(100L); // sleep for 100 ms
```

Comment:
I am thinking if we can (or need to) avoid this 100 ms sleep. I am thinking that it could add another 100ms of latency to queries in some cases.